documentation/memory/step3/exercice/script.js

```diff
@@ -0,0 +1,45 @@
+// node.js dependecies
+import os from 'os';
+import { createHmac } from 'crypto';
+import fs from 'fs/promises';
+
+// constants
+let len = 1_000_000;
+const entries = new Set();
+// fix: const fileName = `entries-${Date.now()}`;
```

Comment:
I don't get the idea of the `fix: *`. In theory, this script should only provide a constant trace report, so the users can analyse and fix it by themselves. Well, you certainly can provide the solution, but you should be careful...

```js
const entries = new Set();
// fix: const fileName = `entries-${Date.now()}`;
``` 
It seems the `new Set()` is a mistake, but its usage is completely fine. I'd mention the `fix:` just in the `entries.add()`. Something like:

```js
  entries.add(entry); // Danger: this will store in the memory: sizeof(entry) * len
```

Reply:
`fix:` comment is just a memo for me, I'll delete them.

My idea is:
- have a script provide a constant trace report
- explain logs in the post
- have a second script without a memory leak, using fs for example.
- readers can observe the difference in the trace report

WDYT?

----------------------------

Also, if I understand your concern correctly, I should remind to readers that the usage of `Set` it's not a bad practice itself? It's more the way we use it that can push pressure on memory.

Reply:
> 

> `fix:` comment is just a memo for me, I'll delete them.
> 
> My idea is:
> 
> * have a script provide a constant trace report
> * explain logs in the post
> * have a second script without a memory leak, using fs for example.
> * readers can observe the difference in the trace report
> 
> WDYT?

:+1: I like it.

> Also, if I understand your concern correctly, I should remind to readers that the usage of Set it's not a bad practice itself? It's more the way we use it that can push pressure on memory.

Yes, that's correct